### ISSUE TITLE
Tensorflow 관련 문의입니다.

### ISSUE THREAD

**Issue by Biyong-Jeong** [DNRY/dlopt#2]
안녕하세요.
딥러닝을 위한 최적화와 수치해석 독자입니다.
시작하여 열심히 하다 보니 텐서프로부분이 처음 실행 안되어 구글링하니
아마 책이 tf 1.0 버전이고 저는 2.0으로 설치한거 같습니다.
2.0 버전으로 실행하고 싶으면 어떻게 해야 하는지와 책을 어떻게 이용할지 문의드립니다.

예를 들면 session / run 만해도 실행이 안됩니다. ㅠ

**Comment by yoon-gu** [comment]
안녕하세요.

텐서플로 1.x과 2.x은 큰 차이가 있습니다.

텐서플로 1.x에서는 자유롭게 loss function 이나 model 들을 customizing 하여 최적화 이론을 적용할 수 있습니다. (하지만, 코드가 길어지고 표현을 처음 접하시는 분들이 어렵습니다.)

이런 것을 보완해주는 keras라는 package가 있습니다. 좀 더 high-level 패키지다보니, 코드도 간결하고 사용하고 처음 접하시는 분들이 쉽게 이해할 수 있습니다. (하지만, model과 loss function의 customizing은 쉽지 않습니다.)

텐서플로 2.x는 사실 케라스의 컨셉을 받아들였습니다. 따라서 session과 run이 필요없는 새로운 접근으로 버전업을 했습니다. 하지만 텐서플로 1.x의 flexibility는 여전히 강력하고, 많이 사용하고 있습니다. 그 예로, 텐서플로 공식 홈페이지에도 모든 문서를 2.x로 대치하는 것이 아닌, 1.x과 2.x의 API들을 모두 소개하고 있습니다.

처음 시작하시는 분들은 텐서플로 2.x가 쉽게 다가 올 수 있으나, 딥러닝 모델에 대한 더 깊은 이해 혹은 최적화 이론이 어떻게 쓰이는지 궁금하신 분들은 텐서플로 1.x가 더 좋은 패키지입니다.

도서에서 소개된 텐서플로 1.x의 내용들을 이해하시면 자연스레 PyTorch,KERAS, 텐서플로 2.x 등의 딥러닝 패키지들을 쉽게 이해할 수 있으실 겁니다.

따라서, 도서에서 소개하는 설치방법을 따라 텐서플로 1.10을 깔고 학습을 진행하시면, 텐서플로 2.x로 딥러닝을 시작하셨을때보다 훨씬 더 깊은 이해를 하실 수 있습니다.
(개인적으로 TF 1.x와 TF 2.x는 Python 2와 Python 3의 변화만큼 많은 변화가 있어서, 상당기간 공존할 가능성이 높다고 생각합니다.)

답변이 도움이 되었길 바라며, 또 추가적인 문의 사항이 있으시면 언제든 알려주세요.

감사합니다.

**Comment by Biyong-Jeong** [comment]
네. 감사합니다.
친절히 설명해주셔서 왜 1.x 로 해야 하는지 이해했습니다.
저는 단순히 2.x로 버전업했으니 버전업 된 것으로 하는게 맞지 않나라는 단순한 생각에
궁금하여 여쭤 보았던 것이고, 답변으로 왜 그런지 이해했습니다.

지금 해당 전공도 아니었고 하는 일도 전혀 딥러닝 등과 무관합니다.
허나 관심이 생기고 관련 이직하고 싶어 공부하는 도중에 좋은 책을 접하여 도움이 많이 되고 있습니다.
걱정이 공부하고 나서 실제 업무에 어떻게 적용하고 관련하여 이직하려면 무엇을 해야 할지 감이 안서고
미리 고민되고 걱정스럽습니다. 
본 도서와 외적인 질문인데 관련하여도 간략한 조언 부탁드려도 될런지요?

책과 관련된 알려주신 사항으로는 계속해서 공부하고 
관련 문의있으면 바쁘시겠지만 질문드리겠습니다.

감사합니다.

**Comment by yoon-gu** [comment]
도움이 되었다니 다행이고 감사합니다!

관련한 문의사항 언제든지 연락주셔도 됩니다 :)
